spacy/tests/matcher/test_phrase_matcher.py

```diff
@@ -323,3 +323,17 @@ def test_phrase_matcher_deprecated(en_vocab):
 @pytest.mark.parametrize("attr", ["SENT_START", "IS_SENT_START"])
 def test_phrase_matcher_sent_start(en_vocab, attr):
     _ = PhraseMatcher(en_vocab, attr=attr)  # noqa: F841
+
+
+def test_span_in_phrasematcher(en_vocab):
+    """Ensure that PhraseMatcher accepts Span and Doc as input"""
+    doc = Doc(en_vocab,
+              words=["I", "like", "Spans", "and", "Docs", "in", "my", "input", ",", "and", "nothing", "else", "."])
+    span = doc[:8]
+    pattern = Doc(en_vocab, words=["Spans", "and", "Docs"])
+    matcher = PhraseMatcher(en_vocab)
+    matcher.add("SPACY", [pattern])
+    matches_doc = matcher(doc)
+    matches_span = matcher(span)
+    assert len(matches_doc) == 1
+    assert len(matches_span) == 1
```

Comment:
Thanks for adding a few tests! I think these are great. Perhaps you could also add one that has multiple matches in the original `doc`, say 3 matches, and then you try and match against a `span` that would only include the middle one? Just to ensure that we're not matching (by accident) outside of the span's boundaries.

Reply:
Added another test (`test_span_v_doc_in_phrasematcher`).  
See comment below.

Reply:
Looks good!